src/PersonalDataSelection.php

```diff
@@ -48,15 +48,21 @@ public function add(string $nameInDownload, array | string $content): PersonalDa
         return $this;
     }
 
-    public function addFile(string $pathToFile, string $diskName = null)
+    public function addFile(string $pathToFile, string $diskName = null, string $folder = null)
     {
         return is_null($diskName)
-            ? $this->copyLocalFile($pathToFile)
-            : $this->copyFileFromDisk($pathToFile, $diskName);
+            ? $this->copyLocalFile($pathToFile, $folder)
+            : $this->copyFileFromDisk($pathToFile, $diskName, $folder);
     }
 
-    protected function copyLocalFile(string $pathToFile)
+    protected function copyLocalFile(string $pathToFile, string $folder = null)
     {
+        if (is_string($folder) && !empty($folder)) {
```

Comment:
This is unreadable to me. Extract this to a function with a good name

Reply:
I refactored it